File: api/Models/Receipt.cs
using System;
using System.Collections.Generic;

namespace api.Models;

public partial class Receipt
{
    public int ReceiptId { get; set; }

    public DateOnly Date { get; set; }

    public decimal TotalAmount { get; set; }

    public decimal Discount { get; set; }

    public decimal TotalValue { get; set; }

    public string ReceiptName { get; set; } = null!;

    public virtual ICollection<ReceiptDetail> ReceiptDetails { get; set; } = new List<ReceiptDetail>();
}

File: vue-project/src/types/Receipt.ts
export interface Receipt {
  receiptId: number,
  receiptName: string,
}
export interface ReceiptDetailInRow {
  quantity: number,
  unit: string,
  price: number,
  totalValue: number,
}

export interface ReceiptSearch extends Receipt {

  totalValue: number
  receiptDetails: ReceiptDetailSearch[]
}

export interface ReceiptDetailSearch {
  instanceName: string
  quantity: number,
  price: number,
  totalValue: number,
}

export interface ReceiptDetail extends ReceiptDetailInRow {
  isInstance: boolean,
  instanceName: string,
}

export interface ReceiptDetailData extends ReceiptDetailInRow {
  instanceName: string | null,
  instanceId: number | null,
  newInstance: string | null,
}

export interface ReceiptResponse {
  receiptName: string,
  date: string,
  totalAmount: number,
  discount: number,
  totalValue: number,
  receiptDetail: ReceiptDetail[]
}

export interface ReceiptData {
  date: string,
  totalAmount: number,
  discount: number,
  totalValue: number,
  receiptDetail: ReceiptDetailData[]
}



